pkg/systemd/journal.js

```diff
@@ -17,152 +17,544 @@
-        if (filler)
-            filler.stop();
-    }
-
-    function update_query() {
-        stop_query();
+    var module = { };
 
-        var match = [ ];
+    /**
+     * journal([match, ...], [options])
+     * @match: any number of journal match strings
+     * @options: an object containing further options
+     *
+     * Load and (by default) stream journal entries as
+     * json objects. This function returns a jQuery deferred
+     * object which delivers the various journal entries.
+     *
+     * The various @match strings are journalctl matches.
+     * Zero, one or more can be specified. They must be in
+     * string format, or arrays of strings.
+     *
+     * The optional @options object can contain the following:
+     *  * "host": the host to load journal from
+     *  * "count": number of entries to load and/or pre-stream.
+     *    Default is 10
+     *  * "follow": if set to false just load entries and don't
+     *    stream further journal data. Default is true.
+     *  * "directory": optional directory to load journal files
+     *  * "boot": when set only list entries from this specific
+     *    boot id, or if null then the current boot.
+     *  * "since": if specified list entries since the date/time
+     *  * "until": if specified list entries until the date/time
+     *  * "cursor": a cursor to start listing entries from
+     *  * "after": a cursor to start listing entries after
+     *
+     * Returns a jQuery deferred promise. You can call these
+     * functions on the deferred to handle the responses. Note that
+     * there are additional non-jQuery methods.
+     *
+     *  .done(function(entries) { }): Called when done, @entries is
+     *         an array of all journal entries loaded. If .stream()
+     *         has been invoked then @entries will be empty.
+     *  .fail(funciton(ex) { }): called if the operation fails
+     *  .stream(function(entries) { }): called when we receive entries
+     *         entries. Called once per batch of journal @entries,
+     *         whether following or not.
+     *  .stop(): stop following or retrieving entries.
+     */
 
-        var query_prio = cockpit.location.options['prio'] || "3";
-        var prio_level = parseInt(query_prio, 10);
-        $("#journal-prio button").each(function() {
-            var num = parseInt($(this).attr("data-prio"), 10);
-            $(this).toggleClass('active', isNaN(prio_level) || num <= prio_level);
-        });
+    module.journalctl = function journalctl(/* ... */) {
+        var matches = [];
+        var options = { follow: true };
+        for (var i = 0; i < arguments.length; i++) {
+            var arg = arguments[i];
+            if (typeof arg == "string") {
+                matches.push(arg);
+            } else if (typeof arg == "object") {
+                if (arg instanceof Array)
+                    matches.push.apply(matches, arg);
+                else
+                    $.extend(options, arg);
```

Comment:
Done, although none of this code was rewritten in this pull request. And checked all the callers.